internal/scheduler/scheduling_algo_test.go

```diff
@@ -677,3 +680,35 @@ func TestLegacySchedulingAlgo_TestSchedule_ExecutorOrdering(t *testing.T) {
 		})
 	}
 }
+
+func BenchmarkNodeDbConstruction(b *testing.B) {
+	for e := 1; e <= 4; e++ {
+		numNodes := int(math.Pow10(e))
+		b.Run(fmt.Sprintf("%d nodes", numNodes), func(b *testing.B) {
+			jobs := testfixtures.N1CpuJobs("queue-alice", testfixtures.PriorityClass0, 32*numNodes)
+			nodes := testfixtures.N32CpuNodes(numNodes, testfixtures.TestPriorities)
+			for i, node := range nodes {
+				for j := 32 * i; j < 32*(i+1); j++ {
+					jobs[j] = jobs[j].WithNewRun("executor-01", node.Name)
+				}
+			}
+			rand.Shuffle(len(jobs), func(i, j int) { jobs[i], jobs[j] = jobs[j], jobs[i] })
```

Comment:
I'm shuffling this so that grouping `jobs` by node (which we didn't have to do before) is slightly harder; I haven't actually checked if it makes a big difference.